### PR TITLE
Fixes to Date Types, manipulation of Resources

### PR DESCRIPTION
**Pull request by vbartusevicius**
* Resource::parentResource property added
* ApiDefinition::removeResource() method added
* `DateTimeType` renamed to `DatetimeType`
* TimeOnlyType format fixed to `H:i:s` instead of `HH:II:SS` 

* fixes to few merge fails

### CODE REVIEW

File: src/ApiDefinition.php [hunk 1]
```diff
@@ -716,6 +716,16 @@ public function addResource(Resource $resource)
         $this->resources[$resource->getUri()] = $resource;
     }
 
+    /**
+     * Removes Resource from ApiDefinition
+     *
+     * @param \Raml\Resource $resource
```
Comment (martin-georgiev) [inline src/ApiDefinition.php]: @param type should be simply `Resource` here as namespace it is already imported.
I also don't think that this docblock adds value. It doesn't provide any information that is not already available from the method's signature.

File: src/Resource.php [hunk 2]
```diff
@@ -82,6 +82,11 @@ class Resource implements ArrayInstantiationInterface
      */
     private $traits = [];
 
+    /**
+     * @var resource|null
```
Comment (martin-georgiev) [inline src/Resource.php]: Correct type here is `Resource`

File: src/Resource.php [hunk 3]
```diff
@@ -455,4 +461,24 @@ public function addTrait($trait)
 
         return $this;
     }
+
+    /**
+     * @return resource|null
+     */
```
Comment (martin-georgiev) [inline src/Resource.php]: Correct type here is `Resource`

File: src/Resource.php [hunk 4]
```diff
@@ -455,4 +461,24 @@ public function addTrait($trait)
...
+    public function getParentResource()
+    {
+        return $this->parentResource;
+    }
+
+    /**
+     * @param resource $parentResource
```
Comment (martin-georgiev) [inline src/Resource.php]: Correct type here is `Resource`

**Comment by vbartusevicius** [comment]
@martin-georgiev - well I executed your suggested command `composer fix-code-style` and got errors about missing binary. I think project is missing it's dev-dependency. I am thinking to add `composer download-php-cs-fixer` to `post-install-cmd` and `post-update-cmd` to always have `php-cs-fixer` downloaded.

**Comment by martin-georgiev** [comment]
@vbartusevicius Apologies for the initially bad instructions as they presume `composer install` will be run prior to `composer fix-code-style`'s execution. 
`composer install` will download the dev packages you're missing and will also place some executables in `bin/` under the project's root. `php-cs-fixer` will be one of those executables.
You probably already installed a previous version of the dependencies either via composer or manually. Therefore it is possible that you may have to delete your local `bin/` and `vendor/` to ensure `composer install` applies the correct locked versions.

**Comment by vbartusevicius** [comment]
@martin-georgiev - removed `bin` dir, but `composer install` did not downloaded `php-cs-fixer`. 
Sadly I cannot see anything in `composer.json` that should download `php-cs-fixer`.

File: src/Types/DatetimeType.php [hunk 5]
```diff
@@ -6,9 +6,9 @@
 use Raml\Type;
 
 /**
- * DateTimeType type class
+ * DatetimeType type class
  */
-class DateTimeType extends Type
+class DatetimeType extends Type
```
Comment (martin-georgiev) [inline src/Types/DatetimeType.php]: Why is this class renaming necessary?

**Comment by martin-georgiev** [comment]
@vbartusevicius Try removing both `bin/` and `vendor/`. Are you sure all dev dependencies are properly installed (do you see folder `vendor/php-cs-fixer`)?

**Comment by vbartusevicius** [comment]
@martin-georgiev - sorry, seems I had old `composer.lock` which stopped from installing `php-cs-fixer`. All good now. Thanks again.

File: src/Types/DatetimeType.php (hunk 5 above)
Comment (martin-georgiev) [reply src/Types/DatetimeType.php]: Ah, right, good spot! I'm surprised this was not reported as a problem earlier.

File: test/ParseTest.php [hunk 6]
```diff
@@ -917,53 +917,4 @@ public function shouldParseResourcePathNameCorrectly()
         $this->assertEquals('Get a single bazDisplayname', $bazId->getDescription());
         $this->assertEquals('Get a list of quxDisplayname', $qux->getDescription());
     }
-
-    /** @test */
-    public function shouldParseTraits()
```
Comment (martin-georgiev) [inline test/ParseTest.php]: Why is this test removed?

File: src/ApiDefinition.php [hunk 7]
```diff
@@ -709,13 +709,23 @@ public function getResources()
...
     }
 
+    /**
+     * Removes Resource from ApiDefinition
+     *
+     * @param Resource $resource
+     */
+    public function removeResource(Resource $resource)
+    {
+        unset($this->resources[$resource->getUri()]);
```
Comment (martin-georgiev) [inline src/ApiDefinition.php]: This needs a safety net checking that the array element is presented.

File: test/ParseTest.php (hunk 6 above)
Comment (vbartusevicius) [reply test/ParseTest.php]: Sorry, had a lot to merge. This slipped through.

**Comment by martin-georgiev** [comment]
@vbartusevicius I became aware that there is an [issue with PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2678) and certain class names. This is affecting `raml-php-parser` CI checks for PHP 7.2, which explains the unexpected casing change from `Resource` to `resource` in one of your commits. Shortly I'll merge several CI and static analysis changes, one of which addresses indeed the above problem. You'll likely have to merge back `master` in your branch later today to inherit the fix.

**Pull request merged**